### ISSUE TITLE
[Bug] reports_model.render_report() does not sanitize input properly

### ISSUE THREAD

**Issue by joshuabuildsthings** [OpenBB-finance/OpenBBTerminal#5299]
**Describe the bug**
When running a report that uses time-series data from Yahoo Finance to pull Futures data like ^GSPC for analysis, the report fails to load because the caret("^") is a disallowed character in file name. nbconvert fails to create the HTML file, because the caret is stripped out of the report_name passed to to the notebook.

**To Reproduce**
Run the default Equity report using ^GSPC as the ticker. It will run successfully and will create the .ipynb, but will not create the HTML report even though the Terminal tells you it did.

**Additional context**
Just update [this line](https://github.com/OpenBB-finance/OpenBBTerminal/blob/develop/openbb_terminal/reports/reports_model.py#L258) in reports_model.create_output_path() to replace carets the way periods('.') are replaced in the file name.

**Comment by deeleeramone** [comment]
The issue that blocks the creation of the report might be more related to the fact that traditional stock items, like financial statements and analyst coverage, do not exist as indices.  Reports are not meant to be "one-size-fits-all" and this is one specific reason why that is the case. An Index is not an Equity.

![Screenshot 2023-08-08 at 8 27 40 PM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/6562a918-e399-4ada-b500-3959a5de9a1d)

![Screenshot 2023-08-08 at 8 28 25 PM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/4958a573-9814-4245-9b1f-e0173ee3e665)

**Comment by joshuabuildsthings** [comment]
Thank you for the quick & thoughtful reply. It looks like you've discovered a separate potential bug! I used the Equity Report as a proxy for a custom report. I should have confirmed the error message first.

Here's a screenshot of the error that occurs; you'll see the caret is stripped when it is passed from the path set by reports_model.create_output_path()  when it is passed by Jupyter.

![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/6468591/9ba79722-3042-4e9d-b7b8-50c9eef4bef0)

Here's the injected parameters to show that what's being set by reports_model.create_output_path() includes the caret:
![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/6468591/7fc13d77-efa4-4c6f-8916-d3178794c727)

**Comment by deeleeramone** [comment]
> Here's a screenshot of the error that occurs; you'll see the caret is stripped when it is passed from the path set by reports_model.create_output_path() when it is passed by Jupyter.

I'm not sure that I follow you here, because if the carat was a point of failure it would always fail.  Here it is on Windows, to show that it is not a Mac or Windows issue.

![Screenshot 2023-08-08 211344](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/4c26a13c-78d4-4d18-81e4-4e9b32b5295c)

You're not really giving any actionable information work with, or a way to recreate, so it's impossible to know what's happening.

**Comment by joshuabuildsthings** [comment]
Thank you for the quick reply. [forecast.ipynb](https://github.com/OpenBB-finance/OpenBBTerminal/blob/develop/openbb_terminal/reports/templates/forecast.ipynb) does not use nbconvert to generate the HTML:
![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/6468591/c4421dab-f1ed-4194-8f66-834b9b49cf85)

As noted in the the initial bug report, as well as the error message in the supplied image; the issue appears to be related to nbconvert's inability to process the path generated by reports_model.create_output_path() due to the caret character.

The function does not sanitize other commonly known [reserved characters](https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file) either:
![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/6468591/2bbda25d-c2ee-4a1b-a80d-3e15b2ef83c3)

A proper path to replication might be:

- Create a Jupyter Notebook using the OpenBB SDK to retrieve stock data via openbb.stocks.load() and output it as a table or graph.

- The Ticker should be set by a command line parameter when run from the OpenBB Terminal. via the reports/run command.

- The last line in the report should use the following: `!jupyter nbconvert {report_name + ".ipynb"} --to html --no-input`

- Put that Jupyter notebook in the user directory.

- Run the Report using OpenBB Terminal, setting the Ticker parameter to ^GSPC.

Hopefully this helps!

**Comment by deeleeramone** [comment]
```
The last line in the report should use the following: !jupyter nbconvert {report_name + ".ipynb"} --to html --no-input
```
Where are you getting this from? The last line of any report is:

```
# to save the results
with open(report_name + ".html", "w", encoding="utf-8") as fh:
    fh.write(report)
```

Which generates the HTML report file, and "^" is not an issue.

![Screenshot 2023-08-08 at 10 35 57 PM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/a001874d-9715-49d9-9484-cdcc97ab62e9)

**Comment by joshuabuildsthings** [comment]
`!jupyter nbconvert {report_name + ".ipynb"} --to html --no-input` is a pretty common code snippet in the Jupyter world.

It's also referenced in the [Reports Documentation](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/openbb_terminal/reports) for OBB:
![image](https://github.com/OpenBB-finance/OpenBBTerminal/assets/6468591/3d2ceefe-339c-4872-8825-bf3ff6a52fb2)

It's likely that the legacy reports used this technique.

The current OBB reports are highly customized and not necessarily reflective of common use cases, while `nbconvert` is absolutely a common tool.

Basically, I see two issues here:
- Code that doesn't properly sanitize special or reserved characters when writing to the file system.
- A bug caused by that, leading to an nbconvert incompatibility.

**Comment by deeleeramone** [comment]
> It's also referenced in the [Reports Documentation](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/openbb_terminal/reports) for OBB: ![image](https://user-images.githubusercontent.com/6468591/259923794-3d2ceefe-339c-4872-8825-bf3ff6a52fb2.png)

This piece is outdated and should be removed.  The reports are generated correctly in the terminal, and this is not the method used.  The carets, as I have demonstrated, are not a point of failure when you use the code supplied within the notebooks.
